src/libraries/Common/src/System/IO/Archiving.Utils.Windows.cs

```diff
@@ -42,5 +43,47 @@ internal static string SanitizeEntryFilePath(string entryPath)
             // There weren't any characters to sanitize.  Just return the original string.
             return entryPath;
         }
+
+        public static unsafe string EntryFromPath(ReadOnlySpan<char> path, bool appendPathSeparator = false)
+        {
+            // Remove leading separators.
+            int nonSlash = path.IndexOfAnyExcept('/', '\\');
+            if (nonSlash == -1)
+            {
+                nonSlash = path.Length;
+            }
+            path = path.Slice(nonSlash);
+
+            // Replace \ with /, and append a separator if necessary.
+
+            if (path.IsEmpty)
+            {
+                return appendPathSeparator ?
+                    "/" :
+                    string.Empty;
+            }
+
+            fixed (char* pathPtr = &MemoryMarshal.GetReference(path))
```

Comment:
Can we simplify it like this:
```suggestion
            fixed (char* pathPtr = path)
```

Reply:
We _can_. It's not exactly the same; there's an extra IsEmpty check with your proposed variant.